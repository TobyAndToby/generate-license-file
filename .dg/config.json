{
  "version": "0.1.0",
  "project": "generate-license-file",
  "outputDir": ".dg",
  "casts": []
}