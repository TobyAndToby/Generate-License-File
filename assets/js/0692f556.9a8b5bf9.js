"use strict";(self.webpackChunk_generate_license_file_website=self.webpackChunk_generate_license_file_website||[]).push([[896],{3185:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>a,contentTitle:()=>s,default:()=>h,frontMatter:()=>r,metadata:()=>c,toc:()=>l});var i=t(5893),o=t(1151);const r={sidebar_position:0,title:"Introduction",description:"Introduction to generate license file"},s=void 0,c={id:"intro/index",title:"Introduction",description:"Introduction to generate license file",source:"@site/versioned_docs/version-3.0.0/intro/index.md",sourceDirName:"intro",slug:"/intro/",permalink:"/docs/intro/",draft:!1,unlisted:!1,editUrl:"https://github.com/TobyAndToby/generate-license-file/tree/main/website/versioned_docs/version-3.0.0/intro/index.md",tags:[],version:"3.0.0",sidebarPosition:0,frontMatter:{sidebar_position:0,title:"Introduction",description:"Introduction to generate license file"},sidebar:"docs",next:{title:"Getting Started",permalink:"/docs/intro/getting-started"}},a={},l=[{value:"What is Generate License File?",id:"what-is-generate-license-file",level:2},{value:"Why should I generate my license file?",id:"why-should-i-generate-my-license-file",level:2},{value:"How can I use it?",id:"how-can-i-use-it",level:2}];function d(e){const n={a:"a",blockquote:"blockquote",code:"code",h2:"h2",li:"li",ol:"ol",p:"p",strong:"strong",ul:"ul",...(0,o.a)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(n.h2,{id:"what-is-generate-license-file",children:"What is Generate License File?"}),"\n",(0,i.jsxs)(n.p,{children:["Generate license file provides tooling to generate a single ",(0,i.jsx)(n.code,{children:".txt"})," file containing all of the licenses for your production third party dependencies. With both a CLI and a programmatic API, you have complete control over how and when the third party license file is generated."]}),"\n",(0,i.jsxs)(n.p,{children:["Generate license file does this by recursively locating the license for each dependency found in the package.json ",(0,i.jsx)(n.code,{children:"dependencies"})," array - starting with a package.json that you provide the path to (most commonly the project root, ",(0,i.jsx)(n.code,{children:"./package.json"}),")."]}),"\n",(0,i.jsx)(n.h2,{id:"why-should-i-generate-my-license-file",children:"Why should I generate my license file?"}),"\n",(0,i.jsx)(n.p,{children:"Third party libraries likely play an integral part of the projects you develop. Most of these libraries will have licenses that govern how that code can be distributed, and how the original authors are attributed to. Under most licenses of the third party software that you use, it is mandatory to distribute their license as part of your software:"}),"\n",(0,i.jsx)(n.p,{children:(0,i.jsx)(n.strong,{children:"MIT"})}),"\n",(0,i.jsxs)(n.blockquote,{children:["\n",(0,i.jsx)(n.p,{children:"The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software..."}),"\n"]}),"\n",(0,i.jsx)(n.p,{children:(0,i.jsx)(n.strong,{children:"ISC"})}),"\n",(0,i.jsxs)(n.blockquote,{children:["\n",(0,i.jsx)(n.p,{children:"Permission to use, copy, modify, and/or distribute this software for any purpose with or without fee is hereby granted, provided that the above copyright notice and this permission notice appear in all copies..."}),"\n"]}),"\n",(0,i.jsx)(n.p,{children:(0,i.jsx)(n.strong,{children:"BSD-3 Clause"})}),"\n",(0,i.jsxs)(n.blockquote,{children:["\n",(0,i.jsx)(n.p,{children:"Redistributions in binary form must reproduce the above copyright notice, this list of conditions..."}),"\n"]}),"\n",(0,i.jsx)(n.p,{children:(0,i.jsx)(n.strong,{children:"LGPLv3"})}),"\n",(0,i.jsxs)(n.blockquote,{children:["\n",(0,i.jsx)(n.p,{children:"For a Combined Work that displays copyright notices during execution, include the copyright notice for the Library..."}),"\n"]}),"\n",(0,i.jsx)(n.p,{children:(0,i.jsx)(n.strong,{children:"Apache 2.0"})}),"\n",(0,i.jsxs)(n.blockquote,{children:["\n",(0,i.jsxs)(n.ol,{start:"4",children:["\n",(0,i.jsxs)(n.li,{children:["\n",(0,i.jsx)(n.p,{children:"Redistribution. You may reproduce and distribute copies of the Work or Derivative Works thereof in any medium, with or without modifications, and in Source or Object form, provided that You meet the following conditions:"}),"\n"]}),"\n"]}),"\n",(0,i.jsxs)(n.ul,{children:["\n",(0,i.jsx)(n.li,{children:"You must give any other recipients of the Work or Derivative Works a copy of this License; and..."}),"\n"]}),"\n"]}),"\n",(0,i.jsx)(n.p,{children:"These are only a fraction of the different types of licenses your third party dependencies may be using. Generate license file automates the process of ensuring you are adhering to your dependencies' licenses."}),"\n",(0,i.jsx)(n.h2,{id:"how-can-i-use-it",children:"How can I use it?"}),"\n",(0,i.jsxs)(n.p,{children:["Head on over to our ",(0,i.jsx)(n.a,{href:"intro/getting-started",children:"Getting Started"})," page to see how you can use generate license file in your projects!"]})]})}function h(e={}){const{wrapper:n}={...(0,o.a)(),...e.components};return n?(0,i.jsx)(n,{...e,children:(0,i.jsx)(d,{...e})}):d(e)}},1151:(e,n,t)=>{t.d(n,{Z:()=>c,a:()=>s});var i=t(7294);const o={},r=i.createContext(o);function s(e){const n=i.useContext(r);return i.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function c(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(o):e.components||o:s(e.components),i.createElement(r.Provider,{value:n},e.children)}}}]);