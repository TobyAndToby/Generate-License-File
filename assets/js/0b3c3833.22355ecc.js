"use strict";(self.webpackChunk_generate_license_file_website=self.webpackChunk_generate_license_file_website||[]).push([[234],{876:(e,t,n)=>{n.d(t,{Zo:()=>p,kt:()=>g});var r=n(2784);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function a(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?a(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):a(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function l(e,t){if(null==e)return{};var n,r,i=function(e,t){if(null==e)return{};var n,r,i={},a=Object.keys(e);for(r=0;r<a.length;r++)n=a[r],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);for(r=0;r<a.length;r++)n=a[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var c=r.createContext({}),s=function(e){var t=r.useContext(c),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},p=function(e){var t=s(e.components);return r.createElement(c.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},d=r.forwardRef((function(e,t){var n=e.components,i=e.mdxType,a=e.originalType,c=e.parentName,p=l(e,["components","mdxType","originalType","parentName"]),d=s(n),g=i,f=d["".concat(c,".").concat(g)]||d[g]||u[g]||a;return n?r.createElement(f,o(o({ref:t},p),{},{components:n})):r.createElement(f,o({ref:t},p))}));function g(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var a=n.length,o=new Array(a);o[0]=d;var l={};for(var c in t)hasOwnProperty.call(t,c)&&(l[c]=t[c]);l.originalType=e,l.mdxType="string"==typeof e?e:i,o[1]=l;for(var s=2;s<a;s++)o[s]=n[s];return r.createElement.apply(null,o)}return r.createElement.apply(null,n)}d.displayName="MDXCreateElement"},4475:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>c,contentTitle:()=>o,default:()=>u,frontMatter:()=>a,metadata:()=>l,toc:()=>s});var r=n(7896),i=(n(2784),n(876));const a={sidebar_position:1,title:"Getting Started",description:"Introductory examples for getting generate license file installed and running"},o=void 0,l={unversionedId:"intro/getting-started",id:"version-2.0.0/intro/getting-started",title:"Getting Started",description:"Introductory examples for getting generate license file installed and running",source:"@site/versioned_docs/version-2.0.0/intro/getting-started.md",sourceDirName:"intro",slug:"/intro/getting-started",permalink:"/docs/intro/getting-started",draft:!1,editUrl:"https://github.com/TobyAndToby/generate-license-file/tree/main/projects/website/versioned_docs/version-2.0.0/intro/getting-started.md",tags:[],version:"2.0.0",sidebarPosition:1,frontMatter:{sidebar_position:1,title:"Getting Started",description:"Introductory examples for getting generate license file installed and running"},sidebar:"docs",previous:{title:"Introduction",permalink:"/docs/intro/"},next:{title:"CLI",permalink:"/docs/cli/"}},c={},s=[{value:"Installation",id:"installation",level:2},{value:"CLI Quick Guide",id:"cli-quick-guide",level:2},{value:"Library Quick Guide",id:"library-quick-guide",level:2}],p={toc:s};function u(e){let{components:t,...n}=e;return(0,i.kt)("wrapper",(0,r.Z)({},p,n,{components:t,mdxType:"MDXLayout"}),(0,i.kt)("p",null,"A couple of quick 5 minute guides for generating your license files via the CLI, or programmatic API."),(0,i.kt)("h2",{id:"installation"},"Installation"),(0,i.kt)("p",null,"Install generate license file via npm:"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-sh"},"npm install --save-dev generate-license-file\n")),(0,i.kt)("h2",{id:"cli-quick-guide"},"CLI Quick Guide"),(0,i.kt)("p",null,"Generate a third party licenses file for the project in the current working directory, outputting to a file called ",(0,i.kt)("inlineCode",{parentName:"p"},"third-party-licenses.txt"),":"),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-bash"},"npx generate-license-file --input ./package.json --output ./third-party-licenses.txt\n")),(0,i.kt)("p",null,"For more advanced usages of the CLI, including the different options and arguments supported, please see the ",(0,i.kt)("a",{parentName:"p",href:"../cli"},"CLI docs"),"."),(0,i.kt)("h2",{id:"library-quick-guide"},"Library Quick Guide"),(0,i.kt)("p",null,"Our library APIs allow you to programmatically interact with generate license file's functionality. Examples include generating and writing the license file to disk, or fetching the license data as an array for other usages. Want to know what the license objects contain? Head over to our ",(0,i.kt)("a",{parentName:"p",href:"../library/api"},"API spec"),"."),(0,i.kt)("pre",null,(0,i.kt)("code",{parentName:"pre",className:"language-js"},'const glf = require("generate-license-file");\n\n// Generate and write the license file to disk.\nglf\n  .generateLicenseFile("./package.json", "./third-party-licenses.txt")\n  .then(() => {})\n  .catch(error => {});\n\n// Get an array of licenses for the current project\'s production dependencies.\nglf\n  .getProjectLicenses("./package.json")\n  .then(licenses => {})\n  .catch(error => {});\n')))}u.isMDXComponent=!0}}]);