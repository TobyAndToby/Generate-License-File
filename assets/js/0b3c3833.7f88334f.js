"use strict";(self.webpackChunk_generate_license_file_website=self.webpackChunk_generate_license_file_website||[]).push([[234],{1014:(e,n,t)=>{t.r(n),t.d(n,{assets:()=>c,contentTitle:()=>a,default:()=>u,frontMatter:()=>s,metadata:()=>o,toc:()=>l});var i=t(5893),r=t(1151);const s={sidebar_position:1,title:"Getting Started",description:"Introductory examples for getting generate license file installed and running"},a=void 0,o={id:"intro/getting-started",title:"Getting Started",description:"Introductory examples for getting generate license file installed and running",source:"@site/versioned_docs/version-2.0.0/intro/getting-started.md",sourceDirName:"intro",slug:"/intro/getting-started",permalink:"/docs/intro/getting-started",draft:!1,unlisted:!1,editUrl:"https://github.com/TobyAndToby/generate-license-file/tree/main/website/versioned_docs/version-2.0.0/intro/getting-started.md",tags:[],version:"2.0.0",sidebarPosition:1,frontMatter:{sidebar_position:1,title:"Getting Started",description:"Introductory examples for getting generate license file installed and running"},sidebar:"docs",previous:{title:"Introduction",permalink:"/docs/intro/"},next:{title:"CLI",permalink:"/docs/cli/"}},c={},l=[{value:"Installation",id:"installation",level:2},{value:"CLI Quick Guide",id:"cli-quick-guide",level:2},{value:"Library Quick Guide",id:"library-quick-guide",level:2}];function d(e){const n={a:"a",code:"code",h2:"h2",p:"p",pre:"pre",...(0,r.a)(),...e.components};return(0,i.jsxs)(i.Fragment,{children:[(0,i.jsx)(n.p,{children:"A couple of quick 5 minute guides for generating your license files via the CLI, or programmatic API."}),"\n",(0,i.jsx)(n.h2,{id:"installation",children:"Installation"}),"\n",(0,i.jsx)(n.p,{children:"Install generate license file via npm:"}),"\n",(0,i.jsx)(n.pre,{children:(0,i.jsx)(n.code,{className:"language-sh",children:"npm install --save-dev generate-license-file\n"})}),"\n",(0,i.jsx)(n.h2,{id:"cli-quick-guide",children:"CLI Quick Guide"}),"\n",(0,i.jsxs)(n.p,{children:["Generate a third party licenses file for the project in the current working directory, outputting to a file called ",(0,i.jsx)(n.code,{children:"third-party-licenses.txt"}),":"]}),"\n",(0,i.jsx)(n.pre,{children:(0,i.jsx)(n.code,{className:"language-bash",children:"npx generate-license-file --input ./package.json --output ./third-party-licenses.txt\n"})}),"\n",(0,i.jsxs)(n.p,{children:["For more advanced usages of the CLI, including the different options and arguments supported, please see the ",(0,i.jsx)(n.a,{href:"../cli",children:"CLI docs"}),"."]}),"\n",(0,i.jsx)(n.h2,{id:"library-quick-guide",children:"Library Quick Guide"}),"\n",(0,i.jsxs)(n.p,{children:["Our library APIs allow you to programmatically interact with generate license file's functionality. Examples include generating and writing the license file to disk, or fetching the license data as an array for other usages. Want to know what the license objects contain? Head over to our ",(0,i.jsx)(n.a,{href:"../library/api",children:"API spec"}),"."]}),"\n",(0,i.jsx)(n.pre,{children:(0,i.jsx)(n.code,{className:"language-js",children:'const glf = require("generate-license-file");\n\n// Generate and write the license file to disk.\nglf\n  .generateLicenseFile("./package.json", "./third-party-licenses.txt")\n  .then(() => {})\n  .catch(error => {});\n\n// Get an array of licenses for the current project\'s production dependencies.\nglf\n  .getProjectLicenses("./package.json")\n  .then(licenses => {})\n  .catch(error => {});\n'})})]})}function u(e={}){const{wrapper:n}={...(0,r.a)(),...e.components};return n?(0,i.jsx)(n,{...e,children:(0,i.jsx)(d,{...e})}):d(e)}},1151:(e,n,t)=>{t.d(n,{Z:()=>o,a:()=>a});var i=t(7294);const r={},s=i.createContext(r);function a(e){const n=i.useContext(s);return i.useMemo((function(){return"function"==typeof e?e(n):{...n,...e}}),[n,e])}function o(e){let n;return n=e.disableParentContext?"function"==typeof e.components?e.components(r):e.components||r:a(e.components),i.createElement(s.Provider,{value:n},e.children)}}}]);