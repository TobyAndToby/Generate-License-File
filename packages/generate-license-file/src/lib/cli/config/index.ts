import { ConfigSchema, parseSchema } from "./schema";
import { doesFileExist } from "../../utils/file.utils";
import { ConfigFile, findConfig, loadConfig } from "./configFile";
import { dirname, isAbsolute, join } from "path";

export const loadConfigFile = async (path?: string): Promise<ConfigSchema> => {
  if (path && (await doesFileExist(path))) {
    return loadAndParseConfig(path);
  } else if (path) {
    // TODO Should this be nicer? (yes).
    throw new Error("Config file could not be found.");
  }

  return findAndParseConfig(process.cwd());
};

const findAndParseConfig = async (directory: string): Promise<ConfigSchema> => {
  const configFile = await findConfig(directory);

  return parseConfig(configFile, directory);
};

const loadAndParseConfig = async (filePath: string) => {
  const configFile = await loadConfig(filePath);
  const directory = dirname(filePath);

  return parseConfig(configFile, directory);
};

const parseConfig = (configFile: ConfigFile | undefined, directory: string): ConfigSchema => {
  const config = parseSchema(configFile?.config);

  if (config === undefined) {
    return;
  }

  for (const replacement in config?.replace) {
    const replacementPath = config.replace[replacement];

    if (isAbsolute(replacementPath)) {
      continue;
    }

    const absolutePath = join(directory, replacementPath);
    config.replace[replacement] = absolutePath;
  }

  if (config.append) {
    for (let i = 0; i < config.append.length; i++) {
      const appendixPath = config.append[i];
      if (isAbsolute(appendixPath)) {
        continue;
      }

      const absolutePath = join(directory, appendixPath);
      config.append[i] = absolutePath;
    }
  }

  return config;
};
