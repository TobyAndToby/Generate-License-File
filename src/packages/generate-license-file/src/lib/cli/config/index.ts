import { dirname, isAbsolute, join } from "path";
import { doesFileExist } from "../../utils/file.utils";
import { ConfigFile, findConfig, loadConfig } from "./configFile";
import { ConfigSchema, parseSchema } from "./schema";

export const loadConfigFile = async (path?: string): Promise<ConfigSchema> => {
  if (path && (await doesFileExist(path))) {
    return loadAndParseConfig(path);
  } else if (path) {
    // TODO Should this be nicer? (yes).
    throw new Error("Config file could not be found.");
  }

  return findAndParseConfig(process.cwd());
};

const findAndParseConfig = async (directory: string): Promise<ConfigSchema> => {
  const configFile = await findConfig(directory);

  return await parseConfig(configFile, directory);
};

const loadAndParseConfig = async (filePath: string) => {
  const configFile = await loadConfig(filePath);
  const directory = dirname(filePath);

  return await parseConfig(configFile, directory);
};

const parseConfig = async (
  configFile: ConfigFile | undefined,
  directory: string,
): Promise<ConfigSchema> => {
  const config = parseSchema(configFile?.config);

  if (config === undefined) {
    return;
  }

  for (const replacement in config?.replace) {
    const replacementPath = config.replace[replacement];

    // The replacement value could be multiple things (e.g. file path, or a
    // URL). If it's a file path, then at this stage the CLI is aware of the
    // execution directory and needs to make the path absolute before handing
    // it off to the library implementation. Otherwise, pass the raw replacement
    // value into the library so it can handle it however it wants.
    const absolutePath = join(directory, replacementPath);

    if (await doesFileExist(absolutePath)) {
      config.replace[replacement] = absolutePath;
    }
  }

  if (config.append) {
    for (let i = 0; i < config.append.length; i++) {
      const appendixPath = config.append[i];
      if (isAbsolute(appendixPath)) {
        continue;
      }

      const absolutePath = join(directory, appendixPath);
      config.append[i] = absolutePath;
    }
  }

  return config;
};
