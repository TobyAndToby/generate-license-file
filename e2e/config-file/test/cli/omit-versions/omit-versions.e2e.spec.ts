import { exec } from "child_process";
import fs from "fs/promises";
import { promisify } from "util";
import { output as outputFileName } from "./config.js";

const execAsync = promisify(exec);

describe("cli", () => {

  it("should omit versions when omitVersions is true in config", async () => {
    const configPath = "./test/cli/omit-versions/config.js";

    await execAsync(`npx generate-license-file -c ${configPath}`);

    const result = await fs.readFile(outputFileName, "utf8");

    // Check that the output doesn't contain version numbers
    // Should not contain patterns like "@1.0.0" or "dep-one@1.0.0"
    expect(result).not.toMatch(/@\d+\.\d+\.\d+/);

    // Should contain package names without versions
    expect(result).toMatch(/dep-one/);
    expect(result).toMatchSnapshot();

    await fs.unlink(outputFileName);
  });
});
