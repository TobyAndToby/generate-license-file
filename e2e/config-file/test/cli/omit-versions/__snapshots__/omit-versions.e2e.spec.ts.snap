// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should omit versions when omitVersions is true in config 1`] = `
"This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file

The following npm package may be included in this product:

 - dep-four

This package contains the following license:

# Dep Four

This license file is spelt \`LICENCE\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-one

This package contains the following license:

# Dep One

This license file is spelt \`LICENSE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-three

This package contains the following license:

# Dep Three

This license file is spelt \`LICENSE\`.
This license should be found.

-----------

The following npm packages may be included in this product:

 - dep-two
 - dep-two-duplicate

These packages each contain the following license:

# Dep Two

This license file is spelt \`LICENCE.md\`.
This license should be found.

-----------

The following npm package may be included in this product:

 - dep-five

This package contains the following license:

MIT

-----------

This file was generated with the generate-license-file npm package!
https://www.npmjs.com/package/generate-license-file
"
`;
